Unit tests for ErrorController. Issue#257

Soupaa opened the pull request:
# Description
Unit tests have been written for 4/5 Error handling functions in ErrorController. The last unfinished unit test, which is for Error 404 has not been written, as the correct solution to solve this specific area cannot be identified within the given time constraint.

Therefore this pull request contains these unit tests:
1. Unit test for Error 401 Unauthorized
2. Unit test for Error 403 Forbidden
3. Unit test for Error 400 BadRequest
4. Unit test for Error UnknownError

Fixes #\ Part of Issue #257

## Type of change


- [x] New feature (non-breaking change which adds functionality)



# How Has This Been Tested?

The four unit tests has been written with xUnit. Therefore this issue is mainly about testing the existing function within ErrorController file.

The testing was conducted with the xUnit "Run Test" and "Run All Tests" buttons.


**Development Configuration**

* Flutter version: 2.0.5
* Dart version: 2.12.3

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

Soupaa committed:
Creating a remote branch. No any real progress yet

Soupaa committed:
ErrorControllerTest now has five unit testing methods.
Four out of five tests passed. Testing for 404 error failed.
Require further investigation into it.

Soupaa committed:
Cleaned up some implementations

Soupaa committed:
Skipping the test for 404 error as it fails.

Soupaa committed:
Removing the method for 404 error testing, as it is unused.
Cleaning up the ErrorControllerTest file.

NikolajKrogh commented on GirafRest/appsettings.Development.json:
```diff
@@ -1,6 +1,6 @@
 ﻿{
     "ConnectionStrings": {
-        "DefaultConnection": "server=192.38.56.153;port=3306;userid=root;password=Giraf123;database=giraf;Allow User Variables=True"
+        "DefaultConnection": "server=localhost; port=3306; userid=user; password=password; database=giraf; Allow User Variables=True"
```
Det er nok en fejl at i har sat server connection til localhost og ændret password

NikolajKrogh commented on GirafRest.Test/Controllers/ErrorControllerTest.cs:
```diff
@@ -0,0 +1,164 @@
+            // Arrange
+            int status = StatusCodes.Status401Unauthorized;
+            var mockFeatureCollection = new Mock<IFeatureCollection>();
+            mockFeatureCollection.Setup(f => f.Get<IStatusCodeReExecuteFeature>())
```
f står for IFeatureCollection ikke? Hvis det gør kan det så ikke få et mere sigende navn så som feature?

NikolajKrogh replied:
Kan se det også sker andre steder

Soupaa replied:
Jo, det kigger jeg på hurtigest muligt

Soupaa committed:
Changes to server connection string and naming in ErrorControllerTest
Changing server connection string to default.
And clarify the variable f as feature + small comment.

Soupaa committed:
Clarify f variable

makelblack159 commented:
kan vi få den fixed så den ikke failer integration test ?

makelblack159 commented on GirafRest.Test/Controllers/ErrorControllerTest.cs:
```diff
@@ -0,0 +1,164 @@
+       public void BadRequest400_Test()
+       {
+            // Arrange
+            int status = 0;
```
giver den ikke en status 400 og ikke 0 ?

Soupaa replied:
Jo, det har du ret i. Den er fikset

Soupaa committed:
Changing in Error 400 Unit test and integration test

JonasStjerne commented:
> kan vi få den fixed så den ikke failer integration test ?

Unit test burde ikke have noget med integration test at gøre. Der bliver kigget på integration test en del det her sprint ved jeg. Der sker en fejl i dem ved connection til database.

Soupaa reviewed (commented):
Changed the json back to dev values
